### ISSUE TITLE
(Feature Request) Close TOC Location List on select

### ISSUE THREAD

**Issue by jrebs** [ixru/nvim-markdown#24]
Hi, I'd post this as a discussion if they were enabled on this repo, but they aren't. I was hoping that it would be possible to add a configuration option to have he TOC location list automatically close when I hit enter on a section item in that list, popping me back to the markdown file at the selected section header. I tried digging around in the plugin source to see how that might be possible so I could contribute a patch myself, but I don't really know the first thing about plugin development for vim. Any chance of seeing this feature added?

**Comment by ixru** [comment]
Yes I see how this could be useful. There's the other usecase of discovery too though, where you want to keep it open to jump around. Maybe even keep the cursor in it when you jump, which it doesn't do currently. I think something like the 'gO' binding from vim docs could be nice. Then you do the keybind if you want the quick jump that closes the window, and :Toc if you want it to stay open. [example using treesitter for implementation maybe](https://github.com/neovim/neovim/blob/214ce8d33c11f75b3500212258f09b58b3d42e80/runtime/lua/vim/vimhelp.lua#L34)  

You can do `autocmd FileType qf nmap <buffer> <cr> <cr>:lcl<cr>` if you want a quick fix, but it will mess with all loclist buffers. If you want to implement it, it only involves making a gO [keymap](https://github.com/ixru/nvim-markdown/blob/f8f8e4191677ccda257ad7ff73a8a36324e0a134/ftplugin/markdown.vim#L788) that calls the [toc function](https://github.com/ixru/nvim-markdown/blob/f8f8e4191677ccda257ad7ff73a8a36324e0a134/ftplugin/markdown.vim#L628) and adds the same keymap as the autocmd.

**Comment by n-crespo** [comment]
Here's one solution that works if you already have a keymap for `:Toc`. If you use `:Toc` i guess you could replace the keymaps setting with `vim.api.nvim_create_user_command()` and do something similar. Also I guess this doesn't work if you wanna use anything toc-related other than `:Toc`. So definitely not a solution that covers every case. But good enough for some.



```lua
-- inside of nvim-markdown config function
vim.api.nvim_create_autocmd("FileType", {
  pattern = "markdown",
  callback = function()
    vim.keymap.set("n", "<leader>cs", function()
      vim.cmd([[Toc]])
      vim.opt_local.number = false -- remove this if you want
      vim.opt_local.relativenumber = false -- remove this if you want
      vim.keymap.set("n", "<CR>", "<cr>:lcl<cr>", { buffer = true })
    end, { buffer = true, silent = true, desc = "Table of Contents (Symbols)" })
  end,
})
```
